### PR TITLE
Fix issues when using file:// and ood together ##debug

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

**Comment by radare** [comment]
Fixes https://github.com/radare/radare2/issues/13391

but this is kind of a workaround, ideally io and bin should differentiate uri and name properly because htats a bit inconsistent

File: libr/core/cmd_open.c [hunk 1]
```diff
@@ -888,6 +886,13 @@ R_API void r_core_file_reopen_debug(RCore *core, const char *args) {
 		(void)r_core_cmd0 (core, "oo");
 		return;
 	}
+	bool is_file_uri = r_str_startswith (binpath, "file://");
+	if (is_file_uri) {
+		eprintf ("Trimming down the 'file://' URI handler.\n");
```
Comment (xarkes) [inline libr/core/cmd_open.c]: Is it really useful to add a warning message? A comment is enough imho, a developer might care while the end user probably not.

File: libr/main/radare2.c [hunk 2]
```diff
@@ -1119,6 +1119,7 @@ R_API int r_main_radare2(int argc, char **argv) {
 								if (r.file && iod && (iod->fd == r.file->fd) && iod->name) {
 									filepath = iod->name;
 								}
+								// XXX if using file:// iod->uri and iod->name are swapped :(
```
Comment (xarkes) [inline libr/main/radare2.c]: Can't you swap them?

File: libr/core/cmd_open.c (hunk 1 above)
Comment (radare) [reply libr/core/cmd_open.c]: uhm yeah better to remove this

File: libr/main/radare2.c (hunk 2 above)
Comment (radare) [reply libr/main/radare2.c]: its fixed

**Pull request merged**